File: Python/TREE/Preorder Traversal.py
//https://www.geeksforgeeks.org/problems/preorder-traversal/1
class Solution:
#Function to return a list containing the preorder traversal of the tree.
    def preorder(self, root):
        result = []

        def traversal(node):
            if node is None:
                return
            result.append(node.data)     # Visit root
            traversal(node.left)         # Traverse left subtree
            traversal(node.right)        # Traverse right subtree

        traversal(root)
        return result

File: Cpp/TREE/Preorder Traversal.cpp
//https://www.geeksforgeeks.org/problems/preorder-traversal/1
class Solution {
  public:

    // Function to return a list containing the preorder traversal of the tree.
    void traverse(Node*root,vector<int>&ans){
        if(root == NULL){
            return;
        }
        ans.push_back(root->data);
  
        traverse(root->left,ans);
        traverse(root->right,ans);
    }
    vector<int> preorder(Node* root) {
        vector<int>ans;
        traverse(root,ans);
        return ans;
       
    }
};
